keras_cv/layers/preprocessing/jittered_resize.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+import keras_cv
+from keras_cv import bounding_box
+from keras_cv import layers
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+
+
+class JitteredResize(BaseImageAugmentationLayer):
+    """JitteredResize implements MaskRCNN style image augmentation.
+
+    JitteredResize takes a three step approach to spatial image augmentation.  This
+    technique is highly optimized for object detection pipelines and was first
+    proposed in the MaskRCNN publication.  The layer takes an input of images and
+    bounding boxes, both of which may be ragged.  It outputs a dense image tensor that
+    is ready to fed to a model for training.
+
+    The augmentation process is as follows:
+
+    The image is first scaled according to a randomly sampled scale factor.  A subset of
```

Comment:
maybe elaborate on this: by scale, do we mean it will respect the aspect ratio? Does it maintain a shorter edge dimension, or longer edge dimension, etc